actionpack/lib/abstract_controller/caching.rb

```diff
@@ -57,7 +57,7 @@ def view_cache_dependencies
       # Convenience accessor.
       def cache(key, options = {}, &block) # :doc:
         if cache_configured?
-          cache_store.fetch(ActiveSupport::Cache.expand_cache_key(key, :controller), options, &block)
+          cache_store.fetch([:controller, key], options, &block)
```

Comment:
This sounds like a behavioural change

Reply:
Nope: in previous implementation `controller` as namespace prepends the key just in the same way as if the key has two parts in current implementation.

Reply:
What about the ENV value?